R/evaluate.R

```diff
@@ -134,34 +134,49 @@ evaluate <- function(input,
   # The user's condition handlers have priority over ours
   handlers <- c(user_handlers, evaluate_handlers)
   
-  for (tle in tles) {
-    watcher$push_source(tle$src, tle$exprs)
-    if (debug || log_echo) {
-      cat_line(tle$src, file = stderr())
-    }
+  cb <- function() {
```

Comment:
I wonder if we should pull this back out into a top level function. It would need arguments `tles`, `log_source`, `watcher`, `handlers`, and `envir`.

Reply:
That would probably be cleaner but no strong feeling either way.

Reply:
Might do in a future PR.